media-libs/embree/embree-4.3.2.ebuild

```diff
@@ -0,0 +1,167 @@
+HOMEPAGE="https://github.com/embree/embree"
+SRC_URI="https://github.com/embree/embree/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="$(ver_cut 1)"
+KEYWORDS="-* ~amd64 ~arm64"
+X86_CPU_FLAGS=( sse2 sse4_2 avx avx2 avx512dq )
+CPU_FLAGS=( cpu_flags_arm_neon "${X86_CPU_FLAGS[@]/#/cpu_flags_x86_}" )
+IUSE="compact-polys ispc +raymask ssp +tbb test ${CPU_FLAGS[*]}"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	amd64? ( || ( ${X86_CPU_FLAGS[*]/#/cpu_flags_x86_} ) )
+	arm64? ( cpu_flags_arm_neon )
+"
+
+BDEPEND="
+	virtual/pkgconfig
+"
+RDEPEND="
+	ispc? ( dev-lang/ispc )
+	tbb? ( dev-cpp/tbb:= )
+"
+DEPEND="${RDEPEND}"
+
+DOCS=( CHANGELOG.md README.md readme.pdf )
+
+PATCHES=(
+	"${FILESDIR}/embree-4.3.1-dont-install-tutorials.patch"
+)
+
+pkg_setup() {
+	CONFIG_CHECK="~TRANSPARENT_HUGEPAGE"
+	WARNING_TRANSPARENT_HUGEPAGE="Not enabling Transparent Hugepages (CONFIG_TRANSPARENT_HUGEPAGE) will impact rendering performance."
+
+	linux-info_pkg_setup
+}
+
+src_prepare() {
+	cmake_src_prepare
+
+	# disable RPM package building
+	sed -e 's|CPACK_RPM_PACKAGE_RELEASE 1|CPACK_RPM_PACKAGE_RELEASE 0|' \
+		-i CMakeLists.txt || die
+
+	# raise cmake minimum version to silence warning
+	sed -e 's#CMAKE_MINIMUM_REQUIRED(VERSION 3.[0-9].0)#CMAKE_MINIMUM_REQUIRED(VERSION 3.5)#I' \
+	    -i \
+	        CMakeLists.txt \
+	        kernels/rthwif/CMakeLists.txt \
+	        tutorials/embree_info/CMakeLists.txt \
+	        tutorials/minimal/CMakeLists.txt \
+	    || die
+}
+
+src_configure() {
+	# -Werror=odr
+	# https://bugs.gentoo.org/859838
+	# https://github.com/embree/embree/issues/481
+	filter-lto
+
+	# NOTE: You can make embree accept custom CXXFLAGS by turning off
+	# EMBREE_IGNORE_CMAKE_CXX_FLAGS. However, the linking will fail if you use
+	# any "m*" compile flags. This is because embree builds modules for the
+	# different supported ISAs and picks the correct one at runtime.
+	# "m*" will pull in cpu instructions that shouldn't be in specific modules
+	# and it fails to link properly.
+	# https://github.com/embree/embree/issues/115
+
+	filter-flags -m*
+
+	# https://bugs.gentoo.org/910164
+	tc-is-clang && filter-lto
+
+	local mycmakeargs=(
+		-DCMAKE_SKIP_INSTALL_RPATH:BOOL=ON
+
+		# Default culling settings for Blender
+		# (Cycles will not render correctly without these)
+		# Some Embree tests will fail with these settings though...
+		-DEMBREE_BACKFACE_CULLING=OFF
+		-DEMBREE_BACKFACE_CULLING_CURVES=ON
+		-DEMBREE_BACKFACE_CULLING_SPHERES=ON
+
+		-DEMBREE_COMPACT_POLYS=$(usex compact-polys)
+
+		# Make sure that we are using our custom compilie flags
+		-DEMBREE_IGNORE_CMAKE_CXX_FLAGS=OFF
+
+		# Set to NONE so we can manually switch on ISAs below
+		-DEMBREE_MAX_ISA:STRING="NONE"
+		-DEMBREE_ISA_AVX=$(usex cpu_flags_x86_avx)
+		-DEMBREE_ISA_AVX2=$(usex cpu_flags_x86_avx2)
+		-DEMBREE_ISA_AVX512=$(usex cpu_flags_x86_avx512dq)
+		# TODO look into neon 2x support
+		-DEMBREE_ISA_NEON=$(usex cpu_flags_arm_neon)
```

Comment:
`cpu_flags_arm_neon` is for 32-bit NEON for `arm` not `arm64`. All arm64 CPUs out there have 64-bit NEON support, and if that's what this is about, you should be just unconditionally enabling NEON on arm64, not having a REQUIRED_USE on a USE-expand flag that is use.masked on arm64.

Something like `$(usex arm64)` should work (untested).

Reply:
That change doesn't work because you are now not able to enable `cpu_flags_arm_neon` on `arm`.  You'd need Something like  `-DEMBREE_ISA_NEON="$(usex arm64 ON "$(usex cpu_flags_arm_neon)")"` but that is a not so pretty solution.

Reply:
I don't think embree supports 32bit CPUs on Linux. So I don't think regular ARM is needed?
(Notice that only 64bit is supported on Mac and Linux here: https://github.com/RenderKit/embree?tab=readme-ov-file#supported-platforms)